controllers/snapshot/snapshot_adapter.go

```diff
@@ -523,9 +528,14 @@ func (a *Adapter) getExistingEnvironmentForScenario(integrationTestScenario *v1b
 }
 
 // createEnvironmentBindingForScenario creates SnapshotEnvironmentBinding for the given test scenario and snapshot
-func (a *Adapter) createEnvironmentBindingForScenario(integrationTestScenario *v1beta1.IntegrationTestScenario, environment *applicationapiv1alpha1.Environment) (*applicationapiv1alpha1.SnapshotEnvironmentBinding, error) {
-	scenarioLabelAndKey := map[string]string{gitops.SnapshotTestScenarioLabel: integrationTestScenario.Name}
-	binding, err := a.createSnapshotEnvironmentBindingForSnapshot(a.application, environment, a.snapshot, scenarioLabelAndKey)
+func (a *Adapter) createEnvironmentBindingForScenario(integrationTestScenario *v1beta1.IntegrationTestScenario, environment *applicationapiv1alpha1.Environment, scenarioOpts ScenarioOptions) (*applicationapiv1alpha1.SnapshotEnvironmentBinding, error) {
+	extraLabels := map[string]string{gitops.SnapshotTestScenarioLabel: integrationTestScenario.Name}
+
+	if scenarioOpts.IsReRun {
```

Comment:
Why we add this label only for the binding created for rerun?

Reply:
to differentiate between SEBs created for reruns so we can skip registering metric for reruns

Reply:
It makes sense. Thanks.